modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/affinity/PartitionExtractor.java

```diff
@@ -366,4 +379,171 @@ else if (rightParam != null)
+
+        if (left instanceof GridSqlOperation) {
+            leftOp = (GridSqlOperation)left;
+            leftOpType = leftOp.operationType();
+
+            if (leftOp.child() instanceof GridSqlColumn) {
+                leftCol = leftOp.child();
+
+                if (!(leftCol.column().getTable() instanceof GridH2Table))
+                    return null;
+            }
+            else
+                return null;
+        }
+        else
+            return null;
+
+        // Try parse right AST.
+        GridSqlAst right = op.child(1);
+        GridSqlOperation rightOp;
+        GridSqlOperationType rightOpType;
+        GridSqlColumn rightCol;
+
+        if (right instanceof GridSqlOperation) {
+            rightOp = (GridSqlOperation)right;
+            rightOpType = rightOp.operationType();
+
+            if (rightOp.child() instanceof GridSqlColumn) {
+                rightCol = rightOp.child();
+
+                if (!(rightCol.column().getTable() instanceof GridH2Table))
+                    return null;
+            }
+            else
+                return null;
+        }
+        else
+            return null;
+
+        // Check that both left and right AST use same column.
+        if (!leftCol.equals(rightCol))
+            return null;
+
+        // Check that columns might be used for partition prunning.
+        if (!((GridH2Table)leftCol.column().getTable()).isColumnForPartitionPruning(leftCol.column()))
+            return null;
+
+        // Check that left AST operation is '>' or '>=' and right AST operation is '<' or '<='.
+        if (!((leftOpType == GridSqlOperationType.BIGGER || leftOpType == GridSqlOperationType.BIGGER_EQUAL)
+            && (rightOpType == GridSqlOperationType.SMALLER || rightOpType == GridSqlOperationType.SMALLER_EQUAL)))
+            return null;
+
+        // Try parse left AST right value (value to the right of '>' or '>=').
+        GridSqlAst leftVal = leftOp.child(1);
+
+        GridSqlConst leftConst;
+        GridSqlParameter leftParam;
+
+        if (leftVal instanceof GridSqlConst) {
+            leftConst = (GridSqlConst)leftVal;
+            leftParam = null;
+        }
+        else if (leftVal instanceof GridSqlParameter) {
+            leftConst = null;
+            leftParam = (GridSqlParameter)leftVal;
+        }
+        else
+            return null;
+
+        // Try parse right AST right value (value to the right of '<' or '<=').
+        GridSqlAst rightVal = rightOp.child(1);
+
+        GridSqlConst rightConst;
+        GridSqlParameter rightParam;
+
+        if (rightVal instanceof GridSqlConst) {
+            rightConst = (GridSqlConst)rightVal;
+            rightParam = null;
+        }
+        else if (rightVal instanceof GridSqlParameter) {
+            rightConst = null;
+            rightParam = (GridSqlParameter)rightVal;
+        }
+        else
+            return null;
+
+        if (leftParam != null || rightParam != null)
```

Comment:
it looks strange. Before we set the parameters in case right(left)Val is instanceOf GridSqlParameter - we can just return null before or this logic is incorrect

Reply:
comment added